src/main/scala/com/mongodb/spark/rdd/partitioner/MongoPaginationPartitioner.scala

```diff
@@ -27,19 +27,24 @@ private[partitioner] trait MongoPaginationPartitioner {
   private implicit object BsonValueOrdering extends BsonValueOrdering
 
   def calculateSkipPartitions(connector: MongoConnector, readConfig: ReadConfig, partitionKey: String, count: Long, skipValues: Seq[Int]): Seq[BsonValue] = {
-    skipValues.map(i => connector.withCollectionDo(readConfig, { coll: MongoCollection[BsonDocument] =>
-      i >= count match {
-        case true => None
+    skipValues.foldLeft(List.empty[(BsonValue, Int)])({ (results: List[(BsonValue, Int)], skipValue) =>
+      skipValue >= count match {
+        case true => results
         case false =>
-          Option(coll.find()
-            .skip(i)
-            .limit(-1)
-            .projection(Projections.include(partitionKey))
-            .sort(Sorts.ascending(partitionKey))
-            .first())
-            .map(_.get(partitionKey))
+          val (preBsonValue, preSkipValue) = if (results.isEmpty) (new BsonMinKey, 0) else results.head
+          val amountToSkip = skipValue - preSkipValue
+          connector.withCollectionDo(readConfig, { coll: MongoCollection[BsonDocument] =>
+            val newHead: Option[(BsonValue, Int)] = Option(coll.find()
+              .filter(Filters.gte(partitionKey, preBsonValue))
```

Comment:
Are there any tests with a bunch of equal partition keys that demonstrate the behavior?  
Since we're looking for partitions, why use gte and not gt?